example_updates/rollback_update/example-rollback-update.sh

```diff
@@ -0,0 +1,798 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Ensure that getopt starts from first option if ". <script.sh>" was used.
+OPTIND=1
+
+ret_val=0
+
+# Ensure we dont end the user's terminal session if invoked from source (".").
+if [[ $0 != "${BASH_SOURCE[0]}" ]]; then
+    ret='return'
+else
+    ret='exit'
+fi
+
+# Output formatting.
+
+# Log level: 0=debug, 1=info, 2=warning, 3=error, 4=none
+log_level=1
+
+warn() { echo -e "\033[1;33mWarning:\033[0m $*" >&2; }
+
+error() { echo -e "\033[1;31mError:\033[0m $*" >&2; }
+
+header() { echo -e "\e[4m\e[1m\e[1;32m$*\e[0m"; }
+
+bullet() { echo -e "\e[1;34m*\e[0m $*"; }
+
+warn "*************************************************"
+warn "*                    WARNING                    *"
```

Comment:
might be worth repeating in comments near the top of file since I've seen it being missed/ignored by several a production customer integration.